File: algorithms/1_easy/M/03038_maximum_number_of_operations_with_the_same_score_i/solution.py
class Solution:
    def maxOperations(self, nums: List[int]) -> int:
        count = 0
        n = len(nums)
        first_sum = nums[0] + nums[1]

        for i in range(0, n, 2):
            if i + 1 < n:
                if nums[i] + nums[i + 1] != first_sum:
                    break
                count += 1

        return count

File: algorithms/1_easy/M/03038_maximum_number_of_operations_with_the_same_score_i/solution.rs
impl Solution {
    pub fn max_operations(nums: Vec<i32>) -> i32 {
        let len = nums.len();
        let first_sum = nums[0] + nums[1];
        let mut count = 0_i32;

        for i in (0..len).step_by(2) {
            if i + 1 < len && nums[i] + nums[i + 1] == first_sum {
                count += 1;
            } else {
                break;
            }
        }

        count
    }
}
